test/jdk/java/nio/file/Files/CopyProcFile.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.io.IOException;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.UncheckedIOException;
+import java.nio.channels.FileChannel;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.ToLongBiFunction;
+import static java.nio.file.StandardOpenOption.*;
+
+import org.testng.Assert;
+import org.testng.annotations.AfterTest;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+/*
+ * @test
+ * @bug 8293502
+ * @requires (os.family == "linux")
+ * @summary Ensure that copying from /proc/mounts works
+ * @run testng/othervm CopyProcFile
+ */
+public class CopyProcFile {
+    static final String SOURCE = "/proc/mounts";
```

Comment:
I wonder how stable this is. If the mount tables change while the test is running then I assume it will fail.

Reply:
Perhaps just using a small prefix of the mount tables would be more stable.

Reply:
/proc/cpuinfo might work. It is of course possible that it could change but maybe less likely.

Reply:
Thanks. Yes, I doubt CPUs are swapped as often as mounts are changed.

Reply:
Changed to `/proc/cpuinfo` in 959011949525bde73f36a702b59d1e27bec569c6.